packages/aws-cdk/bin/cdk.ts

```diff
@@ -232,30 +234,44 @@ async function initCommandLine() {
       case 'bootstrap':
         // Use new bootstrapping if it's requested via environment variable, or if
         // new style stack synthesis has been configured in `cdk.json`.
-        let useNewBootstrapping = false;
-        if (process.env.CDK_NEW_BOOTSTRAP) {
+        //
+        // In code it's optimistically called "default" bootstrapping but that is in
```

Comment:
Maybe I'm being pessimistic, but I wonder if instead of `legacy` and `default` this should just be `v1` and `v2`, or something similar. It seems inevitable that at some point we'll need a `v2.1` or `v3` template, and it would prevent either awkward naming or more refactoring to ack that up-front.

Reply:
The new/"default" template has a versioning mechanism.

So we could also talk about v0/v4 templates... or I guess about v1, v2.4. Any good ideas there?

Reply:
If legacy/default == unversioned/versioned then I suppose the existing (legacy/default) is acceptable. Within the "default" we can always (later) branch on v4.2 or whatever later down the line.